unistd/dir.c

```diff
@@ -488,8 +500,10 @@ static ssize_t _readlink_abs(const char *path, char *buf, size_t bufsiz)
 	if (msg.o.attr.err != EOK)
 		return SET_ERRNO(msg.o.attr.err);
 
-	if (!S_ISLNK(msg.o.attr.val))
-		return SET_ERRNO(-EINVAL);
+	if (!S_ISLNK(msg.o.attr.val)) {
+		SET_ERRNO(-EINVAL);
+		return S_ISDIR(msg.o.attr.val) ? -EISDIR : -1;
```

Comment:
ugly API (returning side info as an error). It's our internal function, maybe add explicit output param (it would increase the readability of the caller code)?

Reply:
I think it's a good idea, I'll change the API a bit to do this more elegantly.